### PR TITLE
Drop support for Node v14

### PR DESCRIPTION
**Pull request by trevor-scheer**

### CODE REVIEW

File: tsconfig.base.json [hunk 1]
```diff
@@ -21,7 +21,8 @@
     "noUnusedLocals": true,
     "noUnusedParameters": true,
     "forceConsistentCasingInFileNames": true,
-    "lib": ["es2019"],
-    "baseUrl": "."
+    "lib": ["es2021"],
+    "baseUrl": ".",
+    "skipLibCheck": true
```
Comment (trevor-scheer) [inline tsconfig.base.json:26]: @glasser this is maybe disappointing to turn on? `lru-cache@8` violates `exactOptionalPropertyTypes` so it triggers compilation errors. Thoughts?

Comment (glasser) [reply tsconfig.base.json:26]: Has this been filed as an issue with them? Also looks like they just went to v9!

File: .changeset/popular-countries-wave.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"@apollo/utils.keyvaluecache": patch
+---
+
+Update lru-cache dependency to v8.x
```
Comment (glasser) [inline .changeset/popular-countries-wave.md:5]: IIUC we use lru-cache's options type as our type so this actually may have user-visible effects... maybe describe in more detail or link to changelog?

File: tsconfig.base.json (hunk 1 above)
Comment (trevor-scheer) [reply tsconfig.base.json:26]: I wouldn't consider it an issue necessarily, it's a configuration option that the project isn't opted in to that we are.

File: .changeset/popular-countries-wave.md (hunk 2 above)
Comment (trevor-scheer) [reply .changeset/popular-countries-wave.md]: Yeah good callout. [476399f](https://github.com/apollographql/apollo-utils/pull/271/commits/476399f86022733b8e0485467b6dc1a7968b83f5)

File: .changeset/popular-countries-wave.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
+---
+"@apollo/utils.keyvaluecache": patch
+---
+
+Update lru-cache dependency to v9.x
```
Comment (glasser) [inline .changeset/popular-countries-wave.md:5]:
```suggestion
Update lru-cache dependency from v7.x to v9.x
```

File: .changeset/popular-countries-wave.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+The `InMemoryLRUCache` constructor accepts an `LRUCache.Options` object. This type has changed since v8.x, so you may need to update your usages accordingly.
+
+The changes in the options typings aren't documented in the changelog, please look at the source below for specifics.
```
Comment (glasser) [inline .changeset/popular-countries-wave.md:9]:
```suggestion
The changes in the options typings aren't documented in the changelog; please look at the source below for specifics.
```

**Review by glasser — APPROVED** [review]

**Pull request merged**